zerver/models/messages.py

```diff
@@ -97,6 +97,8 @@ class MessageType(models.IntegerChoices):
     has_image = models.BooleanField(default=False, db_index=True)
     # Whether the message contains a link.
     has_link = models.BooleanField(default=False, db_index=True)
+    # If the message is a channel message (as opposed to a DM or group-DM)
+    is_channel_message = models.BooleanField(default=True, null=True, db_index=True)
```

Comment:
I checked and `move_rows` in the `zerver.lib.retention` code path copies all fields not specifically excluded, so that should just work.